efel/cppcore/LibV5.cpp

```diff
@@ -157,173 +138,90 @@ int LibV5::ISI_log_slope_skip(mapStr2intVec& IntFeatureData,
-  double inv_first_ISI;
-  vector<double> inv_first_ISI_vec;
-  retVal = getVec(DoubleFeatureData, StringData, "all_ISI_values",
-                        all_isi_values_vec);
-  if (retVal < 1) {
-    inv_first_ISI = 0.0;
-  } else {
-    inv_first_ISI = 1000.0 / all_isi_values_vec[0];
-  }
-  inv_first_ISI_vec.push_back(inv_first_ISI);
-  setVec(DoubleFeatureData, StringData, "inv_first_ISI",
-               inv_first_ISI_vec);
-  return 1;
-}
+  vector<double> last_spike = {peaktime.back() - stimstart[0]};
 
-// 1.0 over second ISI (in Hz); returns 0 when no ISI
-int LibV5::inv_second_ISI(mapStr2intVec& IntFeatureData,
-                          mapStr2doubleVec& DoubleFeatureData,
-                          mapStr2Str& StringData) {
-  int retVal;
-  vector<double> all_isi_values_vec;
-  double inv_second_ISI;
-  vector<double> inv_second_ISI_vec;
-  retVal = getVec(DoubleFeatureData, StringData, "all_ISI_values",
-                        all_isi_values_vec);
-  if (retVal < 2) {
-    inv_second_ISI = 0.0;
-  } else {
-    inv_second_ISI = 1000.0 / all_isi_values_vec[1];
-  }
-  inv_second_ISI_vec.push_back(inv_second_ISI);
-  setVec(DoubleFeatureData, StringData, "inv_second_ISI",
-               inv_second_ISI_vec);
+  setVec(DoubleFeatureData, StringData, "time_to_last_spike", last_spike);
   return 1;
 }
 
-// 1.0 over third ISI (in Hz); returns 0 when no ISI
-int LibV5::inv_third_ISI(mapStr2intVec& IntFeatureData,
-                         mapStr2doubleVec& DoubleFeatureData,
-                         mapStr2Str& StringData) {
-  int retVal;
-  vector<double> all_isi_values_vec;
-  double inv_third_ISI;
-  vector<double> inv_third_ISI_vec;
-  retVal = getVec(DoubleFeatureData, StringData, "all_ISI_values",
-                        all_isi_values_vec);
-  if (retVal < 3) {
-    inv_third_ISI = 0.0;
-  } else {
-    inv_third_ISI = 1000.0 / all_isi_values_vec[2];
+double calculateInvISI(const std::vector<double>& all_isi_values_vec,
+                       size_t index) {
+  if (index < all_isi_values_vec.size()) {
+    return 1000.0 / all_isi_values_vec[index];
   }
-  inv_third_ISI_vec.push_back(inv_third_ISI);
-  setVec(DoubleFeatureData, StringData, "inv_third_ISI",
-               inv_third_ISI_vec);
-  return 1;
+  throw FeatureComputationError("inverse ISI index out of range");
 }
 
-// 1.0 over fourth ISI (in Hz); returns 0 when no ISI
-int LibV5::inv_fourth_ISI(mapStr2intVec& IntFeatureData,
-                          mapStr2doubleVec& DoubleFeatureData,
-                          mapStr2Str& StringData) {
-  int retVal;
-  vector<double> all_isi_values_vec;
-  double inv_fourth_ISI;
-  vector<double> inv_fourth_ISI_vec;
-  retVal = getVec(DoubleFeatureData, StringData, "all_ISI_values",
-                        all_isi_values_vec);
-  if (retVal < 4) {
-    inv_fourth_ISI = 0.0;
-  } else {
-    inv_fourth_ISI = 1000.0 / all_isi_values_vec[3];
+int LibV5::inv_ISI_generic(mapStr2intVec& IntFeatureData,
```

Comment:
If the user just continue using inv_first_ISI, will it call this? Don't you have to modify the Dependency file for that? I'm a little bit confused

Reply:
This beautiful lambda in FillFptrTable.cpp handles it.

```C++
  FptrTableV5["inv_first_ISI"] = [](mapStr2intVec& intData,
                                    mapStr2doubleVec& doubleData,
                                    mapStr2Str& strData) {
    return LibV5::inv_ISI_generic(intData, doubleData, strData, 0);
  };
```

Users keep calling it the same way as in the tests. E.g. `test_basic.py` calls it.